src/developer-quickstart.md

```diff
@@ -0,0 +1,441 @@
+
+For simplicity, we will define the ABI natively in the contract.
+
+```sway
+abi Counter {
+    #[storage(read, write)]
+    fn increment();
+
+    #[storage(read)]
+    fn count() -> u64; 
+}
+```
+
+### Going line by line
+
+`#[storage(read, write)]` is an annotation which denotes that this function has permission to read and write a value in storage.
+
+`fn increment();` - We're introducing the functionality to increment and denoting it shouldn't return any value.
+
+`#[storage(read)]` is an annotation which denotes that this function has permission to read values in storage.
+
+`fn counter() -> u64;` - We're introducing the functionality to increment the counter and denoting the function's return value.
+
+### Implement ABI
+
+Below your ABI definition, you will write the implementation of the functions defined in your ABI.
+
+```sway
+impl Counter for Contract {
+    #[storage(read)]
+    fn count() -> u64 {
+      return storage.counter;
+    }
+    #[storage(read, write)]
+    fn increment(){
+        storage.counter = storage.counter + 1;
+    }
+}
+```
+
+> **Note**
+> `return storage.counter;` is equivalent to `storage.counter`.
+
+### What we just did
+
+Read and return the counter property value from the contract storage.
+
+```sway
+fn count() -> u64 {
+    return storage.counter;
+}
+```
+
+The function body accesses the value counter in storage, and increments the value by one. Then, we return the newly updated value of counter.
+
+```sway
+fn increment() {
+    storage.counter = storage.counter + 1;
+}
+```
+
+Here's what your code should look like so far:
+
+```sway
+contract;
+
+storage {
+    counter: u64 = 0,
+}
+
+abi Counter {
+    #[storage(read, write)]
+    fn increment();
+
+    #[storage(read)]
+    fn count() -> u64;
+}
+
+impl Counter for Contract {
+    #[storage(read)]
+    fn count() -> u64 {
+      return storage.counter;
+    }
+    #[storage(read, write)]
+    fn increment(){
+        storage.counter = storage.counter + 1;
+    }
+}
+```
+
+### Build the Contract
+
+Build `counter_dapp` by running the following command in your terminal from inside the `counter_dapp` directory:
+
+```sh
+forc build
+```
+
+You should see something like this output:
+
+```console
+Compiled library "core".
+  Compiled library "std".
+  Compiled contract "counter_dapp".
+  Bytecode size is 240 bytes.
+```
+
+## Testing your Contract
+
+In the directory `tests`, navigate to `harness.rs.` Here you'll see there is some boilerplate code to help you start interacting with and testing your contract.
+
+At the bottom of the file, define the body of `can_get_contract_instance`. Here is what your code should look like to verify that the value of the counter did get incremented:
+
+```sway
```

Comment:
Maybe add a step to run cargo build before changing harness to have code completion. For example, in my case, I had the rust-analyzer installed, and it started to show some errors on the file.

Reply:
mmm, what do you mean? Before editing the file, the boilerplate was throwing errors? @luizstacio